rllib/execution/parallel_requests.py

```diff
@@ -182,3 +183,224 @@ def wait_asynchronous_requests(
+        self._max_remote_requests_in_flight = max_remote_requests_in_flight
+        self._all_workers = set(workers)
+        self._num_workers = len(workers)
+        self._available_workers = workers.copy()
+        self._available_workers_set = set(self._available_workers)
+        self._unavailable_workers = set()
+        self._remote_requests_in_flight = defaultdict(set)
+        self._pending_to_actor = {}
+        self._call_queue = Queue(maxsize=0)
+        self._pending_remotes = []
+
+    def submit(
+        self,
+        remote_fn: Callable,
+        *,
+        fn_args: List[Any] = None,
+        fn_kwargs: Dict[str, Any] = None,
+        for_all_workers: bool = False,
+    ) -> None:
+        """Submit a remote function call schedule on available workers.
+
+        Args:
+            remote_fn: The remote function to call
+            fn_args: The arguments to pass to the remote function
+            fn_kwargs: The keyword arguments to pass to the remote function
+            for_all_workers: If True, submit this request to the queue for as many
+                number of workers that there are. If False, submit the request without
+                replicating it.
+
+        """
+        if fn_args is None:
+            fn_args = []
+        if fn_kwargs is None:
+            fn_kwargs = {}
+        call = (remote_fn, fn_args, fn_kwargs)
+        if for_all_workers:
+            for _ in range(self._num_workers):
+                self._call_queue.put(call)
+        else:
+            self._call_queue.put(call)
+        self._run()
+
+    def _run(self) -> int:
+        """Run submitted requests on available workers"""
+        num_requests_launched = 0
+        available_actor = self._get_available_actor()
+        while available_actor and self._call_queue.qsize() > 0:
+            remote_fn, fn_args, fn_kwargs = self._call_queue.get()
+            req = available_actor.apply.remote(remote_fn, *fn_args, **fn_kwargs)
+            self._remote_requests_in_flight[available_actor].add(req)
+            self._pending_remotes.append(req)
+            self._pending_to_actor[req] = available_actor
+            available_actor = self._get_available_actor()
+            num_requests_launched += 1
+        return num_requests_launched
+
+    def get_ready_requests(self) -> Dict[ActorHandle, List[Any]]:
+        """Get requests that are ready to be returned
+
+        Returns:
+            A dictionary of actor handles to lists of returns from tasks that were
+             previously submitted to this actor pool that are now ready to be returned.
+             If return_object_refs
+
+        """
+        ready_requests_dict = defaultdict(list)
+        ready_requests, self._pending_remotes = ray.wait(
+            self._pending_remotes,
+            timeout=self._ray_wait_timeout_s,
+            num_returns=len(self._pending_remotes),
+        )
+        if not self._return_object_refs:
+            objs = ray.get(ready_requests)
```

Comment:
do you need to check if ready_requests is empty?

Reply:
You would get an empty list back from `ray.get([])`, so I think it's fine.